### ISSUE TITLE
Proposal Indicate to user when `elastic-package cluster up` is ready?

### ISSUE THREAD

**Issue by ruflin** [elastic/elastic-package#55]
At the moment, when someone runs `elastic-package cluster up` and the command returns, the cluster is not actually ready to use and it can take quite a bit long until Kibana is available. This might be confusing to the user. To debug, `docker ps` has to be used to check the health status.

My proposal would be to only return the up command when the cluster is ready to be worked with. In this test script I implemented a check: https://github.com/elastic/package-storage/blob/production/testing/main_integration_test.go#L54 This could be added to the up command.

**Comment by mtojek** [comment]
Hm.. we may also bring up another fake service that blocks the process until everything (incl. Kibana is healthy). I suppose it would be just additional configuration in `docker-compose.yml`?

**Comment by ruflin** [comment]
Assuming we get `--insecure` working, this additional service could be an agent ;-)

**Comment by mtojek** [comment]
Yes and no ;) After introducing the agent and defining dependency on Kibana, we'll shift the problem from Kibana to the Agent.

**Comment by ruflin** [comment]
The difference here is, that on the agent side we talk about seconds and on the Kibana side about minutes.

**Comment by ycombinator** [comment]
I have two PRs open related to the discussion here:
1. To include a fake `is_ready` service in the cluster's `docker-compose.yml` that will block on the `kibana` service being healthy: #71.
2. A PR to expose the `--insecure` flag from `elastic-agent` as an environment variable: https://github.com/elastic/beats/pull/20713.

I think we should keep the `is_ready` service around as it is super lightweight and can always serve as the final check on whether the cluster/stack is up, no matter what other services are part of the cluster/stack and in what order they are spun up. That is, even after https://github.com/elastic/beats/pull/20713 is resolved and if we choose to add an `elastic-agent` service, it can depend on `kibana` and `is_ready` can depend on `elastic-agent`.
